### PR TITLE
Move common parts of redis connection interface to common

### PR DESCRIPTION
**Pull request by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg**
This is part of the effort to build a proxy for Redis Cluster (#5697).

We want to make some of the connection interface common so we can share it with the cluster proxy. Specifically, we take `redis_proxy/conn_pool.h` and move much of it besides `Instance` to `common/redis/client.h`, and adjust the rest of the project to work with this. The formatting tool did some spacing stuff as well.

Following this, we will want to make an upstream command abstraction (for commands sent by Envoy upstream and that come back and terminate at Envoy) such as the health checks (exist, ping) and cluster (cluster slots).

### CODE REVIEW

**Comment by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg** [comment]
oops, I'll fix those errors

**Comment by mattklein123** [comment]
@HenryYYang for first pass.

**Comment by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg** [comment]
As it turns out, the formatter rearranges imports, and this breaks the build! Hence my testing working locally, as I didn't test post-formatto.

File: source/extensions/filters/network/redis_proxy/conn_pool_impl.h [hunk 1]
```diff
@@ -30,7 +31,7 @@ namespace ConnPool {
 // TODO(mattklein123): Circuit breaking
 // TODO(rshriram): Fault injection
 
-class ConfigImpl : public Config {
+class ConfigImpl : public Common::Redis::Config {
```
Comment (HenryYYang) [inline source/extensions/filters/network/redis_proxy/conn_pool_impl.h]: Should ConfigImpl also be moved to the common library?

Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply source/extensions/filters/network/redis_proxy/conn_pool_impl.h]: sure sounds good

File: test/extensions/filters/network/common/redis/test_utils.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+namespace Client {
+
+inline envoy::config::filter::network::redis_proxy::v2::RedisProxy::ConnPoolSettings
```
Comment (HenryYYang) [inline test/extensions/filters/network/common/redis/test_utils.h]: this was not inline before, right?

**Comment by HenryYYang** [comment]
LGTM

File: test/extensions/filters/network/common/redis/test_utils.h (hunk 2 above)
Comment (mattklein123) [reply test/extensions/filters/network/common/redis/test_utils.h]: This is fine, or you can move the impl to a CC file in a follow up.

Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply test/extensions/filters/network/common/redis/test_utils.h]: It was not- I inlined it as it was short (per the style guide this ok apparently), but it's an easy move if you guys prefer.

**Pull request merged**